examples/gno.land/p/demo/memeland/memeland.gno

```diff
@@ -0,0 +1,196 @@
+package memeland
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Post - as previously defined
+type Post struct {
+	ID            string
+	Data          string
+	Author        std.Address
+	Timestamp     time.Time
+	Upvotes       int
+	UpvoteTracker *avl.Tree // address > bool hasVoted
+}
+
+type Memeland struct {
+	Posts       *avl.Tree
+	MemeCounter seqid.ID
+}
+
+func NewMemeland() *Memeland {
+	return &Memeland{
+		Posts: avl.NewTree(),
+	}
+}
+
+// PostMeme - Adds a new post
+func (m *Memeland) PostMeme(data, timestamp string) string {
+	if data == "" || timestamp == "" {
+		return "timestamp and data cannot be empty"
+	}
+
+	// Generate ID
+	id := m.MemeCounter.Next().String()
```

Comment:
Why not use a library like `seqid`?

Reply:
I actually am using it, take a better look ;)